public/app/features/dashboard/folder_picker/folder_picker.ts

```diff
@@ -132,23 +132,26 @@ export class FolderPickerCtrl {
   }
 
   private loadInitialValue() {
-    if (this.initialFolderId && this.initialFolderId > 0) {
-      this.getOptions('').then(result => {
+    this.getOptions('').then(result => {
+      if (!_.isNil(this.initialFolderId)) {
+        // If initialFolderId is set, try to find it in result or return null
         this.folder = _.find(result, { value: this.initialFolderId });
         if (!this.folder) {
-          this.folder = { text: this.initialTitle, value: this.initialFolderId };
+          this.folder = { text: this.initialTitle, value: null };
         }
-        this.onFolderLoad();
-      });
-    } else {
-      if (this.initialTitle && this.initialFolderId === null) {
-        this.folder = { text: this.initialTitle, value: null };
       } else {
-        this.folder = { text: this.rootName, value: 0 };
+        // If initialFolderId isn't set, return General folder for Editor
+        // or first available for user, otherwise return null
+        if (this.isEditor) {
```

Comment:
Right now these changes breaking the feature of reset implemented in https://github.com/grafana/grafana/pull/11712/files#diff-79eaf83858455a078c9c8f0196b0397eR62

Reply:
Not sure why it brakes it. What should `enableReset` allow to do?

Reply:
In the alert list and dashboard list panels the default folder value is *All*. When I tested and loaded a dashboard with alert list and dashboard list panels with folder of *All* it was set to General in folder picker when inspecting edit mode.

The `enableReset` allows the *All* to be selected in the alert list and dashboard list panels scenario given a valid folder is selected before.